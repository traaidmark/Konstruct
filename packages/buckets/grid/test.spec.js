// BUCKET [ GRID > TESTS ] #####################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Grid from './src';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. PROPS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

const props = {
  largeCols: '1fr 1fr',
  largeGutter: 'var(--gutter-medium)',
  mediumCols: '1fr 1fr',
  mediumGutter: 'var(--gutter)',
  smallCols: '1fr',
  smallGutter: 'var(--gutter)',
};

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Bucket: Grid', () => {

  // 3.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Grid>
        <div>col 1</div>  
        <div>col 2</div>  
      </Grid>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

  // 3.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Grid
        smallCols={ props.smallCols }
        smallGutter={ props.smallGutter }
        mediumCols={ props.mediumCols }
        mediumGutter={ props.mediumGutter }
        largeCols={ props.largeCols }
        largeGutter={ props.largeGutter }
      >
        <div>col 1</div>  
        <div>col 2</div>  
      </Grid>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.2. END ..................................................................

});

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
