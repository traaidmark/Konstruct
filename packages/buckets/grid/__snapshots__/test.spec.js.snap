// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bucket: Grid renders correctly with all props 1`] = `
.c0 {
  display: grid;
}

@media (max-width:480px) {
  .c0 {
    grid-template-columns: 1fr;
    grid-gap: var(--gutter);
  }
}

@media (min-width:481px) and (max-width:1023px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter);
  }
}

@media (min-width:1024px) and (max-width:1440px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter-medium);
  }
}

@media (min-width:1441px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter-medium);
  }
}

<div
  className="c0"
>
  <div>
    col 1
  </div>
  <div>
    col 2
  </div>
</div>
`;

exports[`Bucket: Grid renders correctly without any props 1`] = `
.c0 {
  display: grid;
}

@media (max-width:480px) {
  .c0 {
    grid-template-columns: 1fr;
    grid-gap: var(--gutter);
  }
}

@media (min-width:481px) and (max-width:1023px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter);
  }
}

@media (min-width:1024px) and (max-width:1440px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter-medium);
  }
}

@media (min-width:1441px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
    grid-gap: var(--gutter-medium);
  }
}

<div
  className="c0"
>
  <div>
    col 1
  </div>
  <div>
    col 2
  </div>
</div>
`;
